pkg/operator/latencyprofilecontroller/profilerejectionchecker.go

```diff
@@ -0,0 +1,110 @@
+package latencyprofilecontroller
+
+import (
+	"fmt"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	listersv1 "k8s.io/client-go/listers/core/v1"
+
+	configv1 "github.com/openshift/api/config/v1"
+	nodeobserver "github.com/openshift/library-go/pkg/operator/configobserver/node"
+)
+
+type profileRejectRevisionChecker struct {
+	configMapLister           listersv1.ConfigMapNamespaceLister
+	profileRejectionScenarios []nodeobserver.LatencyProfileRejectionScenario
+	latencyConfigs            []nodeobserver.LatencyConfigProfileTuple
+}
+
+// NewInstallerProfileRejectionChecker is used to construct a CheckProfileRejectionFunc that
+// can compare the (installer controller generated config map) config of the latest active revision
+// with the desired/target latency profile to return whether profile transition should be rejected or not.
+// The latencyConfigs passed to this func check arg values of each latency profile to compare
+// them across the active config on the cluster. In case the transition is from profile found in active config
+// to specified target profile and it is for one of the scenarios specified in profileRejectionScenarios,
+// the rejection flag is returned as true with an appropriate message; false otherwise.
+// Eg. transition between extreme latency profiles viz. Default <-> Low should be rejected.
+func NewInstallerProfileRejectionChecker(
+	configMapLister listersv1.ConfigMapNamespaceLister,
+	profileRejectionScenarios []nodeobserver.LatencyProfileRejectionScenario,
+	latencyConfigs []nodeobserver.LatencyConfigProfileTuple,
```

Comment:
let's put the latencyConfigs before the rejectionScenarios to be consistent with other functions

Reply:
Updated the function signature accordingly.